### PR TITLE
UserDefaults CRUD actions for favourites.

### PR DESCRIPTION
**Pull request by steven-hill**
This allows the user to have a list separate from the search results. The user can easily navigate to the list via the tab bar.

It is handled by a Persistence Manager and a Favourites Coordinator.

The “Add to favourites” button title label in MovieDetailVC changes when tapped to tell the user the movie was added to the favourites list. The delete action is shown via swipe to delete on the table view cell in FavouritesListVC.

To be added - an alert to tell the user if the UserDefaults actions fail.

### CODE REVIEW

File: MovieList/MovieList/Favourites/PersistenceManager.swift [hunk 1]
```diff
@@ -9,46 +9,48 @@ import Foundation
...
     }
     
-    func addToFavourites(movie: MovieResult) {
-        let newFavourite = MovieResult(trackName: movie.trackName, artworkUrl100: movie.artworkUrl100, primaryGenreName: movie.primaryGenreName, longDescription: movie.longDescription, releaseDate: movie.releaseDate, previewUrl: movie.previewUrl)
-        
-        guard !favourites.contains(where: { movie in
-            favourites.append(newFavourite)
-            return true
-        }) else { return }
+    func addToFavourites(name: String) {
```
Comment (OlivAlex) [inline MovieList/MovieList/Favourites/PersistenceManager.swift]: Possibly rename to: `func saveMovieNameToFavourites(newMovie: String)`

**Review by OlivAlex — COMMENTED** [review]
LGTM!

File: MovieList/MovieList/Favourites/PersistenceManager.swift (hunk 1 above)
Comment (steven-hill) [reply MovieList/MovieList/Favourites/PersistenceManager.swift]: Renamed to: func saveMovieNameToFavourites(newMovie: String).

**Comment by steven-hill** [comment]
After completing the changes, the app was tested on the simulator. All CRUD actions still work as expected.

**Pull request merged**